src/ucp/wireup/address.c

```diff
@@ -279,24 +262,30 @@ ucp_address_gather_devices(ucp_worker_h worker, ucp_ep_h ep, uint64_t tl_bitmap,
 static size_t ucp_address_packed_size(ucp_worker_h worker,
                                       const ucp_address_packed_device_t *devices,
                                       ucp_rsc_index_t num_devices,
-                                      uint64_t flags)
+                                      uint64_t pack_flags)
 {
     size_t size = 0;
     const ucp_address_packed_device_t *dev;
 
-    if (flags & UCP_ADDRESS_PACK_FLAG_WORKER_UUID) {
+    /* header: version and flags */
+    size += 1;
+
+    if (pack_flags & UCP_ADDRESS_PACK_FLAG_WORKER_UUID) {
         size += sizeof(uint64_t);
     }
 
-    size += ucp_address_worker_name_size(worker, flags);
+    if ((worker->context->config.ext.address_debug_info) &&
+        (pack_flags & UCP_ADDRESS_PACK_FLAG_WORKER_NAME)) {
+        size += strlen(ucp_worker_get_name(worker)) + 1;
```

Comment:
it can be trimmed by max_worker_name, right?

Reply:
it's already trimmed when worker is created